packages/aws-cdk/lib/api/resource-import/importer.ts

```diff
@@ -102,17 +103,15 @@ export class ResourceImporter {
 
   private readonly stack: cxapi.CloudFormationStackArtifact;
   private readonly cfn: Deployments;
-  private readonly ioHost: IIoHost;
-  private readonly action: ToolkitAction;
+  private readonly ioHost: ActionAwareIoHost;
 
   constructor(
     stack: cxapi.CloudFormationStackArtifact,
     props: ResourceImporterProps,
   ) {
     this.stack = stack;
     this.cfn = props.deployments;
-    this.ioHost = props.ioHost;
-    this.action = props.action;
+    this.ioHost = withAction(props.ioHost, props.action as any);
```

Comment:
Almost the future. Eventually this class should just get an `ActionAwareIoHost` in the first place.